src/StackAdmin/Azs.Fabric.Admin/custom/Get-AzsVolume.ps1

```diff
@@ -0,0 +1,187 @@
+  [Location <String>]: Location of the resource.
+  [ResourceGroupName <String>]: Name of the resource group.
+  [ScaleUnit <String>]: Name of the scale units.
+  [StorageSubSystem <String>]: Name of the storage system.
+  [SubscriptionId <String>]: Subscription credentials that uniquely identify Microsoft Azure subscription. The subscription ID forms part of the URI for every service call.
+  [Volume <String>]: Name of the storage volume.
+.Link
+https://docs.microsoft.com/en-us/powershell/module/azs.fabric.admin/get-azsvolume
+#>
+function Get-AzsVolume {
+[OutputType([Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Models.Api20190501.IVolume])]
+[CmdletBinding(DefaultParameterSetName='List', PositionalBinding=$false)]
+param(
+    [Parameter(ParameterSetName='Get')]
+    [Parameter(ParameterSetName='List')]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Category('Path')]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Runtime.DefaultInfo(Script='(Get-AzLocation)[0].Name')]
+    [System.String]
+    # Location of the resource.
+    ${Location},
+
+    [Parameter(ParameterSetName='Get', Mandatory)]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Category('Path')]
+    [System.String]
+    # Name of the storage volume.
+    ${Name},
+
+    [Parameter(ParameterSetName='Get')]
+    [Parameter(ParameterSetName='List')]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Category('Path')]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Runtime.DefaultInfo(Script='-join("System.",(Get-AzLocation)[0].Name)')]
+    [System.String]
+    # Name of the resource group.
+    ${ResourceGroupName},
+
+    [Parameter(ParameterSetName='Get', Mandatory)]
+    [Parameter(ParameterSetName='List', Mandatory)]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Category('Path')]
+    [System.String]
+    # Name of the scale units.
+    ${ScaleUnit},
+
+    [Parameter(ParameterSetName='Get', Mandatory)]
+    [Parameter(ParameterSetName='List', Mandatory)]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Category('Path')]
+    [System.String]
+    # Name of the storage system.
+    ${StorageSubSystem},
+
+    [Parameter(ParameterSetName='Get')]
+    [Parameter(ParameterSetName='List')]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Category('Path')]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Runtime.DefaultInfo(Script='(Get-AzContext).Subscription.Id')]
+    [System.String[]]
+    # Subscription credentials that uniquely identify Microsoft Azure subscription.
+    # The subscription ID forms part of the URI for every service call.
+    ${SubscriptionId},
+
+    [Parameter(ParameterSetName='GetViaIdentity', Mandatory, ValueFromPipeline)]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Category('Path')]
+    [Microsoft.Azure.PowerShell.Cmdlets.FabricAdmin.Models.IFabricAdminIdentity]
+    # Identity Parameter
+    # To construct, see NOTES section for INPUTOBJECT properties and create a hash table.
+    ${InputObject},
```

Comment:
@rakku-ms Did we do something for removing the InputObject parameter in the New * cmdlets?

Reply:
Yes, this can removed using following customization:

## variant removal (parameter InputObject) from all New cmdlets -- parameter sets GetViaIdentity
  - where:
      verb: New
      variant: GetViaIdentity
    remove: true

Reply:
@bganapa For fabric storage admin, InputObject is needed (otherwise, we need the resourceId, refer to your comment in teams below).
[12/13/19 4:25 PM] Bala Ganapathy
    Yuxing Zhou, We dont need the ResourceId parameter, autorest generates a InputObject parameter in all the Get cmdlets where Id is one of the fields. This is the replacement for the ResourceId Parameter

If we need to remove this parameter for all RPs, please let me know and I will confirm with our PM.